src/connections/allContacts.js

```diff
@@ -104,3 +102,18 @@ export const withContactsMutations = withMutations(client => ({
 }))
 
 export default flow([withContacts, withContactsMutations])
+
+export function mergeContact(contact, attributes) {
+  function concat(arr1 = [], arr2 = []) {
+    return [...arr1, ...arr2]
+  }
+  function customizer(objValue, srcValue, key) {
+    if (Array.isArray(objValue) || Array.isArray(srcValue)) {
+      return concat(objValue, srcValue)
+    }
+    if (key === 'fullname') {
+      return objValue.length > srcValue.length ? objValue : srcValue
```

Comment:
Wouldn't this make data loss possible?
E.g. `{fullname: "John J. Doe", ...}` and `{fullname: "John Doe Jr", ...}`? (sorry, I can't think of a better example)

What about checking whether some of the strings include the other one, defaulting to both slash-concatenated? It should work assuming the attributes containers (contacts here) are considered identicals (i.e. with the same identity, not necessarily the same value).

Reply:
(this one sound like a blocker)

Reply:
In https://github.com/cozy/cozy-doctypes/pull/57 I suggest to remove this field.

And I changed the target of `takeLonger` strategy to become `givenName` and `familyName`.